### ISSUE TITLE
errors checking for latest version in AWS Lambda, also on Mac with both Python 2 & Python 3

### ISSUE THREAD

**Issue by cjchapman** [googlefonts/fontbakery#2405]
## Observed behaviour

```
>> com.google.fonts/check/fontbakery_version
   Do we have the latest version of FontBakery installed?
 * ERROR: Failed with TypeError: argument of type 'NoneType' is not iterable
           File "/usr/local/lib/python3.7/site-packages/fontbakery/checkrunner.py", line 345, in _exec_check
             for sub_result in result:  # Might raise.
           File "/usr/local/lib/python3.7/site-packages/fontbakery/specifications/general.py", line 216, in com_google_fonts_check_fontbakery_version
             if not (is_latest or is_up_to_date(installed_str, latest_str)):
           File "/usr/local/lib/python3.7/site-packages/fontbakery/specifications/general.py", line 175, in is_up_to_date
             has_githash = ".dev" in installed
         

   Result: ERROR
```
## Resources and exact process needed to replicate

This is on a colleague's Macintosh with:
```
fontbakery==0.6.12
Python 3.7.2
```

I can't reproduce this on my machine with Python 3.7.0, but thought I should report this as all other checks seem to work correctly on my colleague's machine.

**Comment by cjchapman** [comment]
OK... we've run into this in AWS Lamba and this is what we're getting:
```b"The directory '/home/sbx_user1051/.cache/pip/http' or its parent directory is not owned by the current user and the cache has been disabled. Please check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.\nfontbakery (0.7.1)  - Well designed Font QA tool, written in Python 3\n"```
...where the code was expecting:
```b'fontbakery (0.7.1)  - Well designed Font QA tool, written in Python 3\n  INSTALLED: 0.7.2.dev2+g40067089\n  LATEST:    0.7.1\n'```

**Comment by cjchapman** [comment]
On the colleague's Mac that I mentioned in the original report, we got:
```b"fontbakery (0.7.1)  - Well designed Font QA tool, written in Python 3\nYou are using pip version 9.0.1, however version 19.0.3 is available.\nYou should consider upgrading via the 'pip install --upgrade pip' command.\n"```
On further inspection we found that it was using the pip for Python 2, not the pip for Python 3:
```$ pip --version
pip 19.0.3 from /Library/Python/2.7/site-packages/pip (python 2.7)```

**Comment by cjchapman** [comment]
So, just to be clear, we're experience two different problems with this check: one due to filesystem permissions issues in AWS Lambda, the other due to the fact that the user had both Python 2 & Python 3 installed on her Mac and `pip` was referring to Python 2.

**Comment by cjchapman** [comment]
We've resolved the problem on my colleague's Mac for now by symbolically linking `pip` to `pip3`, but I've got to imagine we're going to run into that problem on other machines, and we still have the AWS Lambda issue, so I think we need to find a more robust way to do this check.

**Comment by cjchapman** [comment]
On the AWS Lambda front, we tried explicitly calling pip3 just to be sure it wasn't a pip2 vs pip3 issue, but we still get:
```
b"The directory '/home/sbx_user1051/.cache/pip/http' or its parent directory is not owned by the current user and the cache has been disabled. Please check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.\nfontbakery (0.7.1)  - Well designed Font QA tool, written in Python 3\n"
```

**Comment by miguelsousa** [comment]
@felipesanches This issue can be closed. I was resolved by #3016
